samples/popup/components/openPopup.js

```diff
@@ -0,0 +1,56 @@
+AFRAME.registerComponent("open-popup", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+  multiple: true,
+  /**
+   * Initial creation and setting of the mesh.
+   */
+  init: function () {
+    const data = this.data.jsonData;
+    const el = this.el;
+
+    // Create geometry.
+    this.geometry = new THREE.BoxBufferGeometry(
+      data.width,
+      data.height,
+      data.depth
+    );
+
+    // Create material.
+    this.material = new THREE.MeshStandardMaterial({ color: data.color });
+
+    // Create mesh.
+    this.mesh = new THREE.Mesh(this.geometry, this.material);
+
+    el.object3D.position.set(data.x, data.y, data.z);
+    // Set mesh on entity.
+    el.setObject3D("mesh", this.mesh);
+    el.addEventListener("loaded", function (e) {
```

Comment:
Sorry, I should have been clearer, I meant we need to listen for the `loaded` on line 55, as `blackboardEl.appendChild(entityEl);` is the call that is async. So, we need to do this in the opposite order.
Like:
```
el.addEventListener("click", function () {
          entityEl = addEntityToBlackboard(JSON.stringify(data.blackboardData));
          // add a listener to entityEl being `loaded` and do lines 37 and 38 in there. 
```
Does that make sense? Let me know if that's confusing

Reply:
That would mean `entityEl` needs to be returned from line 55. Or you could do `// add a listener to entityEl being `loaded` and do lines 37 and 38 in there` inside of there

Reply:
PS, check slack :)